embassy-rp/src/binary_info/mod.rs

```diff
@@ -0,0 +1,174 @@
+//! Code and types for creating Picotool compatible "Binary Info" metadata
+//!
+//! Add something like this to your program, and compile with the "binary-info"
+//! and "rt" features:
+//!
+//! ```
+//! # use rp235x_hal as hal;
```

Comment:
Wrong crate name. This is caught by doctests. Can you make `cargo test --no-default-features --features time-driver,rp235xa` pass, then add it to `.gthub/ci/test.sh`?

Reply:
This turned out to be quite annoying. Cortex-m gets confused so I've hacked around it [here](https://github.com/embassy-rs/embassy/blob/778241fd71b8a0cb541d23b69c3f9e47ec2753ff/embassy-rp/src/lib.rs#L518) but I'm not happy about it. I have to assume there's a smarter way to handle this.